venonactl/pkg/plugins/helper.go

```diff
@@ -130,3 +149,64 @@ func getKubeObjectsFromTempalte(values map[string]interface{}, pattern string, l
 	templatesMap := templates.TemplatesMap()
 	return KubeObjectsFromTemplates(templatesMap, values, pattern, logger)
 }
+
+func ensureClusterRequirements(client *kubernetes.Clientset, req validationRequest) (validationResult, error) {
+	result := validationResult{}
+	nodes, err := client.CoreV1().Nodes().List(metav1.ListOptions{})
+	if err != nil {
+		return result, err
+	}
+	if nodes == nil {
+		return result, errors.New("Nodes not found")
+	}
+
+	result.isValid = true
+	if len(nodes.Items) == 0 {
+		result.message = append(result.message, "No nodes in cluster")
+		result.isValid = false
+	}
+	for _, n := range nodes.Items {
+		res := testNode(n, req)
+		if len(res) > 0 {
+			result.message = append(result.message, res...)
+			result.isValid = false
+		}
+	}
+	if !req.strictMode {
+		result.isValid = true
+	}
+	return result, nil
+}
+
+func testNode(n v1.Node, req validationRequest) []string {
+	result := []string{}
+
+	if req.cpu != "" {
+		requiredCPU, err := resource.ParseQuantity(req.cpu)
+		if err != nil {
+			result = append(result, err.Error())
+			return result
+		}
+		cpu := n.Status.Capacity.Cpu()
+
+		if cpu != nil && cpu.Cmp(requiredCPU) == -1 {
```

Comment:
I find this test too strict, consider a scenario where you have n nodes but only one node doesn't have enough cpu/memory , this shouldn't block the installation. Consider to block installation if all the nodes failed the test